datalad/__init__.py

```diff
@@ -12,6 +12,14 @@
 
 # Other imports are interspersed with lgr.debug to ease troubleshooting startup
 # delays etc.
+
+# If there is a bundled git, make sure GitPython uses it too:
+from datalad.cmd import GitRunner
+_git_runner = GitRunner()
+if _git_runner._GIT_PATH:
+    import os
+    os.environ['GIT_PYTHON_GIT_EXECUTABLE'] = os.path.join(_git_runner._GIT_PATH, 'git')
```

Comment:
hm, ok... for completness 
- indeed there are some rogue imports
```shell
$> git grep -e 'import git *$' -e 'from git import'
datalad/crawler/nodes/annex.py:from git import Repo
datalad/distribution/rewrite_urls.py:    from git import GitConfigParser
datalad/support/annexrepo.py:from git import InvalidGitRepositoryError
datalad/support/tests/test_annexrepo.py:import git
datalad/support/tests/test_annexrepo.py:from git import GitCommandError
datalad/tests/test_testrepos.py:import git
datalad/tests/utils.py:import git
```

my concern is a slight but additional impact on the start time of 'datalad' cmdline tools.  From quick timing measurements it adds ~10-20ms (I guess be cause it needs to sense external git etc) to overall 380ms of running `datalad --help`. Seems to be little but it all adds up and I did spend quite a bit of time trying to cut it down.
So I guess 
- for now it would be ok to add it here indeed
- I wonder if we could add some test if that setting is in effect so later on if I come to optimize things up again, I don't ruin it?
- may instead of initiating unused GitRunner, just call `GitRunner._check_git_path()` -- yeap, it is protected API but at least it would carve off may be a few ns of dealing with possible protocols etc?